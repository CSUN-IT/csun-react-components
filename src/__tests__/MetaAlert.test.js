import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import MetaAlert from '../../src/components/meta/Alert';

configure({ adapter: new Adapter() });

describe('MetaAlert', () => {
  it('Should be defined', () => {
    expect(MetaAlert).toBeDefined();
  });

  it('Renders component correctly', () => {
    const mockFn = jest.fn();
    const tree = shallow(
      <MetaAlert
        color={'success'}
        strongText={'Pro Tip:'}
        text={'Color is success'}
        onClick={mockFn}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
