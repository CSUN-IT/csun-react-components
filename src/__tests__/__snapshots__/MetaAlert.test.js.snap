// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaAlert Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaAlert
    className=""
    color="success"
    onClick={[MockFunction]}
    strongText="Pro Tip:"
    text="Color is success"
    to="#"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <strong>
          Pro Tip:
        </strong>,
        " ",
        "Color is success",
        <a
          className="alert__close"
          data-alert-close={true}
          href="#"
          onClick={[MockFunction]}
        >
          ×
        </a>,
      ],
      "className": " alert alert--success",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pro Tip:",
        },
        "ref": null,
        "rendered": "Pro Tip:",
        "type": "strong",
      },
      " ",
      "Color is success",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "×",
          "className": "alert__close",
          "data-alert-close": true,
          "href": "#",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "×",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <strong>
            Pro Tip:
          </strong>,
          " ",
          "Color is success",
          <a
            className="alert__close"
            data-alert-close={true}
            href="#"
            onClick={[MockFunction]}
          >
            ×
          </a>,
        ],
        "className": " alert alert--success",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pro Tip:",
          },
          "ref": null,
          "rendered": "Pro Tip:",
          "type": "strong",
        },
        " ",
        "Color is success",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "×",
            "className": "alert__close",
            "data-alert-close": true,
            "href": "#",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "×",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
